simpleviewinc/tap-events-force
# ZEN-627: Session Details Modal Updates

**Ticket**: [ZEN-627](https://jira.simpleviewtools.com/browse/zen-627)

## Goal

* Update session details modal layout and positioning styles
* Do some refactoring with `reStyle`

## Updates

* `src/components/modals/session/sessionDetails.js`
  * Renamed the `Body` component to `SessionDetails` and extracted it to its own file
  * Updated styles mentioned in the ticket to match mockup
* `src/components/modals/session/sessionDetailsHeader.js`
  * Extracted header to its own file
  * Updated styles mentioned in the ticket to match mockup
* `src/components/modals/session/sessionDetailsModal.js`
  * just moved this under a `session` folder, within `modals`
* `src/components/sessionDetails/sessionPresenters.js`
  * Simplified
    * Replace `useStyle` with `reStyle` and removed associated theme file styles
    * Removed unused style prop
* `src/constants/values.js`
  * Moved `DISPLAY_PROPERTIES` to `EVF_CATEGORIES`
    * it was previously incorrectly placed under `INTERNAL_CATEGORIES`, which meant it wasn't being read from the root props. This was causing a bug (that went unnoticed) which prevented the Sessions component consumer from changed display properties
* `src/mocks/eventsforce/evfModalBuilder.js`
  * updated the modal to use the `large` size so that we can actually see content properly
* `src/theme/components/modals/sessionDetailsModal.js`
  * removed unused styles (now provided by `reStyle`)


## Testing
* `keg evf pack run package=evf:zen-627-session-details-modal-updates`
  * This runs this branch, with the latest reStyle changes synced in from [this PR](https://github.com/simpleviewinc/keg-hub/pull/199)
* Open up any session details modal
* Verify that the styles match the figma mockup: https://www.figma.com/file/aLesACdNG3CvInUVuFm4hO/Session-selection---final?node-id=0%3A1
* Note that I only focused on updating the features listed in the ticket: 
    * Move the header title down
    * Add vertical divider between time and day-name
    * Ensure there are dividers between description, labels, and top section
    * Move labels to the bottom
    * Move select button to top-right
* Other style updates are handled in other tickets
